library("riboSeqR")
setwd("~/digging/riboseq/")
ribodatadir <- "bam/"
rnadatadir <- "rnaseq_bam/"
#araFasta <- "~/genome/TAIR10_cdna_20110103_representative_gene_model_updated"
fastaCDS <- findCDS(fastaFile = "~/genome/TAIR10_cdna.fa",startCodon = c("ATG"),stopCodon = c("TAG", "TAA", "TGA"))
ribofiles <- paste(ribodatadir, c("RISCR1","RISCR2","RISDR1","RISDR2"),".bam", sep = "")
rnafiles <- paste(rnadatadir, c("RSCR1","RSCR2","RSDR1","RSDR2"),".bam", sep="")
riboDat <- readRibodata(ribofiles,rnafiles,replicates = c("WT", "WT", "ddm1", "ddm1"))
riboDat <- readRibodata(ribofiles,replicates = c("WT", "WT", "ddm1", "ddm1"))
fCs <- frameCounting(riboDat,fastaCDS)
fS <- readingFrame(rC = fCs)
riboCounts <- sliceCounts(ffCs, lengths = c(27, 28), frames = list(1, 0))
x <-cbind(as.data.frame(ffCs@CDS),as.data.frame(riboCounts))
write.table(x,"ffcs",row.names=F,sep="\t",quote=F)
#ffCs <- filterHits(fcl, lengths = c(27),frames = list(1),fS = fsl)
#riboCounts <- sliceCounts(ffCs,lengths = c(27),frames = list(1))
#ffCs <- filterHits(fcl, lengths = c(27,28),frames = list(1,0),fS = fsl)
#riboCounts <- sliceCounts(ffCs,lengths = c(27,28),frames = list(1,0))
#ffCs <- filterHits(fcl,lengths = c(27,28,29),frames = list(1,0,2),fS = fsl)
#riboCounts <- sliceCounts(ffCs,lengths = c(27,28,29),frames = list(1,0,2))
pdf("r1.pdf")riboCounts
plotFS(fS)
dev.off()
save.image()
fcl<- frameCounting(riboDat,fastaCDS,lengths =20:50)
fsl <- readingFrame(rC = fcl,lengths =20:50)
# ffCs <- filterHits(fcl, lengths = c(27, 28,29), frames = list(1, 0,2), hitMean = 50, unqhitMean = 10, fS = fsl)
#ffCs <- filterHits(fcl,lengths = c(27,28,29),frames = list(1,0,2),fS = fsl)
#riboCounts <- sliceCounts(ffCs,lengths = c(27,28,29),frames = list(1,0,2))
setEPS();postscript("riboseq_20-50_1.eps",onefile = FALSE)
plotFS(fsl[[1]]);dev.off()
setEPS();postscript("riboseq_20-50_2.eps",onefile = FALSE)
plotFS(fsl[[2]]);dev.off()
setEPS();postscript("riboseq_20-50_3.eps",onefile = FALSE)
plotFS(fsl[[3]]);dev.off()
setEPS();postscript("riboseq_20-50_4.eps",onefile = FALSE)
plotFS(fsl[[4]]);dev.off()

setEPS();postscript("riboseq_20-45_1.eps",onefile = FALSE)
plotFS(fsl[[1]],lengths =20:45);dev.off()
setEPS();postscript("riboseq_20-45_2.eps",onefile = FALSE)
plotFS(fsl[[2]],lengths =20:45);dev.off()
setEPS();postscript("riboseq_20-45_3.eps",onefile = FALSE)

setEPS();postscript("riboseq_20-45_4.eps",onefile = FALSE)
plotFS(fsl[[4]],lengths =20:45);dev.off()

ffCs <- filterHits(fcl,lengths = c(27,28,29),frames = list(1,0,2),fS = fsl)
riboCounts <- sliceCounts(ffCs,lengths = c(27,28,29),frames = list(1,0,2))
setEPS();postscript("riboseq_27_1.eps",onefile = FALSE)
pdf("riboseq_27.pdf")
plotCDS(coordinates = ffCs@CDS, riboDat = riboDat, lengths = 27)
dev.off()
pdf("riboseq_28.pdf")
plotCDS(coordinates = ffCs@CDS, riboDat = riboDat, lengths = 28)
dev.off()
pdf("riboseq_29.pdf")
pdf("riboseq_29.pdf")
plotCDS(coordinates = ffCs@CDS, riboDat = riboDat, lengths = 29)
dev.off()



setEPS();postscript("evade2.eps")
plotTranscript("AT5G44925.1", coordinates = fcl@CDS,riboData = riboDat, length = 28, cap = 200)
dev.off()

rnafiles <- paste(rnadatadir, c("RSCR1","RSCR2","RSDR1","RSDR2"),"_trans.bam", sep="")
rnaDat <- readRibodata(rnafiles,replicates = c("WT", "WT", "ddm1", "ddm1"))
rcl<- frameCounting(rnaDat,fastaCDS,lengths =20:50)

write.table(as.data.frame(fastaCDS),"fastaCDS",row.names=F,sep="\t",quote=F)
write.table(as.data.frame(fCs@hits),"fCs",row.names=F,sep="\t",quote=F)
library("riboSeqR")
trans <- read.table("~/genome/transposopons_trans_id")
seqnames(fastaCDS) %in% trans$V1

setEPS();postscript("transposon.eps")
> tfastaCDS <- fastaCDS[seqnames(fastaCDS) %in% trans$V1,]
> tfCs <- frameCounting(riboDat,tfastaCDS,lengths =20:50)
> tfastaCDS <- fastaCDS[seqnames(fastaCDS) %in% trans$V1,]
> tfsl <- readingFrame(rC =tfCs,lengths =20:50)
> setEPS();postscript("transposon2.eps")
> plotFS(tfsl[[2]]);dev.off()
> setEPS();postscript("transposon3.eps")
> plotFS(tfsl[[3]]);dev.off()
> setEPS();postscript("transposon4.eps")
> plotFS(tfsl[[4]]);dev.off()
tffCs <- filterHits(tfCs, lengths = c(27,28,29), frames = list(1,0,2),hitMean = 50, unqhitMean = 10, fS = tfsl)
plotCDS(coordinates = tffCs[[1]]@CDS, riboDat = riboDat, lengths = 27)
plotCDS(coordinates = tffCs[[1]]@CDS, riboDat = riboDat, lengths = 28)
plotCDS(coordinates = tffCs[[1]]@CDS, riboDat = riboDat, lengths = 29)

> tfastaCDS <- fastaCDS[seqnames(fastaCDS) %in% trans$V1,]
> tfsl <- readingFrame(rC =tfCs,lengths =20:50)
> setEPS();postscript("transposon2.eps")
> plotFS(tfsl[[2]]);dev.off()
> setEPS();postscript("transposon3.eps")
> plotFS(tfsl[[3]]);dev.off()
> setEPS();postscript("transposon4.eps")
> plotFS(tfsl[[4]]);dev.off()
tffCs <- filterHits(tfCs, lengths = c(27,28,29), frames = list(1,0,2),fS = tfsl)
#plotCDS(coordinates = tffCs[[1]]@CDS, riboDat = riboDat, lengths = 27)
#plotCDS(coordinates = tffCs[[1]]@CDS, riboDat = riboDat, lengths = 28)
#plotCDS(coordinates = tffCs[[1]]@CDS, riboDat = riboDat, lengths = 29)

#ffCs <- filterHits(fcl,lengths = c(27,28,29),frames = list(1,0,2),fS = fsl)
riboCounts <- sliceCounts(tffCs,lengths = c(27,28,29),frames = list(1,0,2))
#setEPS();postscript("transposons_27_1.eps",onefile = FALSE)
pdf("transposons_27.pdf")
plotCDS(coordinates = tffCs@CDS, riboDat = riboDat, lengths = 27)
dev.off()
pdf("transposons_28.pdf")
plotCDS(coordinates = tffCs@CDS, riboDat = riboDat, lengths = 28)
dev.off()
pdf("transposons_29.pdf")
plotCDS(coordinates = tffCs@CDS, riboDat = riboDat, lengths = 29)
dev.off()
